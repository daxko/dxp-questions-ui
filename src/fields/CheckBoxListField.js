var React = require('react');

var CheckBoxListField = React.createClass({

	onChange: function(e) {

		var new_values = (this.props.answer || []).slice(0);

		if (e.target.checked) {
			if (new_values.indexOf(e.target.value) === -1) {
				new_values.push(e.target.value);
			}
		} else {
			var index = new_values.indexOf(e.target.value);
			if (index >= 0) {
				new_values.splice(index, 1);
			}
		}

		this.props.onChange(new_values);
		this.props.onBlur();
	},

	render: function() {

		var question = this.props.question;
		var answer = this.props.answer || [];
		var checked = {};
		for (var i = 0; i < answer.length; i++) {
			checked[answer[i]] = true;
		}

		return (
			<div className="dxp-checkbox">
				{ 
					question.possible_answers.map(function(pa) {
						return (
							<label key={pa.id}>
								<input type="checkbox" 
									name={this.props.question_id} 
									value={pa.id}
									disabled={this.props.question.read_only}
									onChange={this.onChange}
									onBlur={this.props.onBlur}
									checked={checked[pa.id] || false} />
								{pa.display_value}
							</label>
						);
					}.bind(this))
				}
			</div>
		);
	}

});

module.exports = CheckBoxListField;