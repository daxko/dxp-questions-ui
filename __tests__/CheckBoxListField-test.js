import React from 'react'
import {configure, shallow} from 'enzyme'
import Adapter from 'enzyme-adapter-react-15'
import CheckBoxListField from '../src/fields/CheckBoxListField'
const json = require('../snippets/json/checkbox_question.json')

configure({adapter: new Adapter() })

describe('CheckBoxListField component', () => {
  it('should render a question', () => {
    const wrapper = shallow(<CheckBoxListField question={json} />)
    expect(wrapper).toMatchSnapshot()
  })
})