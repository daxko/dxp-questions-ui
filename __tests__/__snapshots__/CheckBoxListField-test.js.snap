// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBoxListField component should render a question 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CheckBoxListField
    question={
      Object {
        "default_value": null,
        "description": "",
        "empty_value": "",
        "hide_empty_option": false,
        "possible_answers": Array [
          Object {
            "amount": null,
            "display_value": "Soccer",
            "id": "childcare_possible_answer-103475",
            "value": "Soccer",
          },
          Object {
            "amount": null,
            "display_value": "Basketball",
            "id": "childcare_possible_answer-103476",
            "value": "Basketball",
          },
          Object {
            "amount": null,
            "display_value": "Swimming",
            "id": "childcare_possible_answer-103477",
            "value": "Swimming",
          },
          Object {
            "amount": null,
            "display_value": "Football",
            "id": "childcare_possible_answer-103478",
            "value": "Football",
          },
        ],
        "read_only": false,
        "required": false,
        "title": "What other sports have you participated in?",
        "type": "checkboxes",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label>
          <input
            checked={false}
            disabled={false}
            name={undefined}
            onBlur={undefined}
            onChange={[Function]}
            type="checkbox"
            value="childcare_possible_answer-103475"
          />
          Soccer
        </label>,
        <label>
          <input
            checked={false}
            disabled={false}
            name={undefined}
            onBlur={undefined}
            onChange={[Function]}
            type="checkbox"
            value="childcare_possible_answer-103476"
          />
          Basketball
        </label>,
        <label>
          <input
            checked={false}
            disabled={false}
            name={undefined}
            onBlur={undefined}
            onChange={[Function]}
            type="checkbox"
            value="childcare_possible_answer-103477"
          />
          Swimming
        </label>,
        <label>
          <input
            checked={false}
            disabled={false}
            name={undefined}
            onBlur={undefined}
            onChange={[Function]}
            type="checkbox"
            value="childcare_possible_answer-103478"
          />
          Football
        </label>,
      ],
      "className": "dxp-checkbox",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "childcare_possible_answer-103475",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              checked={false}
              disabled={false}
              name={undefined}
              onBlur={undefined}
              onChange={[Function]}
              type="checkbox"
              value="childcare_possible_answer-103475"
            />,
            "Soccer",
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "checked": false,
              "disabled": false,
              "name": undefined,
              "onBlur": undefined,
              "onChange": [Function],
              "type": "checkbox",
              "value": "childcare_possible_answer-103475",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "Soccer",
        ],
        "type": "label",
      },
      Object {
        "instance": null,
        "key": "childcare_possible_answer-103476",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              checked={false}
              disabled={false}
              name={undefined}
              onBlur={undefined}
              onChange={[Function]}
              type="checkbox"
              value="childcare_possible_answer-103476"
            />,
            "Basketball",
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "checked": false,
              "disabled": false,
              "name": undefined,
              "onBlur": undefined,
              "onChange": [Function],
              "type": "checkbox",
              "value": "childcare_possible_answer-103476",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "Basketball",
        ],
        "type": "label",
      },
      Object {
        "instance": null,
        "key": "childcare_possible_answer-103477",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              checked={false}
              disabled={false}
              name={undefined}
              onBlur={undefined}
              onChange={[Function]}
              type="checkbox"
              value="childcare_possible_answer-103477"
            />,
            "Swimming",
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "checked": false,
              "disabled": false,
              "name": undefined,
              "onBlur": undefined,
              "onChange": [Function],
              "type": "checkbox",
              "value": "childcare_possible_answer-103477",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "Swimming",
        ],
        "type": "label",
      },
      Object {
        "instance": null,
        "key": "childcare_possible_answer-103478",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              checked={false}
              disabled={false}
              name={undefined}
              onBlur={undefined}
              onChange={[Function]}
              type="checkbox"
              value="childcare_possible_answer-103478"
            />,
            "Football",
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "checked": false,
              "disabled": false,
              "name": undefined,
              "onBlur": undefined,
              "onChange": [Function],
              "type": "checkbox",
              "value": "childcare_possible_answer-103478",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "Football",
        ],
        "type": "label",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label>
            <input
              checked={false}
              disabled={false}
              name={undefined}
              onBlur={undefined}
              onChange={[Function]}
              type="checkbox"
              value="childcare_possible_answer-103475"
            />
            Soccer
          </label>,
          <label>
            <input
              checked={false}
              disabled={false}
              name={undefined}
              onBlur={undefined}
              onChange={[Function]}
              type="checkbox"
              value="childcare_possible_answer-103476"
            />
            Basketball
          </label>,
          <label>
            <input
              checked={false}
              disabled={false}
              name={undefined}
              onBlur={undefined}
              onChange={[Function]}
              type="checkbox"
              value="childcare_possible_answer-103477"
            />
            Swimming
          </label>,
          <label>
            <input
              checked={false}
              disabled={false}
              name={undefined}
              onBlur={undefined}
              onChange={[Function]}
              type="checkbox"
              value="childcare_possible_answer-103478"
            />
            Football
          </label>,
        ],
        "className": "dxp-checkbox",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "childcare_possible_answer-103475",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                checked={false}
                disabled={false}
                name={undefined}
                onBlur={undefined}
                onChange={[Function]}
                type="checkbox"
                value="childcare_possible_answer-103475"
              />,
              "Soccer",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "checked": false,
                "disabled": false,
                "name": undefined,
                "onBlur": undefined,
                "onChange": [Function],
                "type": "checkbox",
                "value": "childcare_possible_answer-103475",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            "Soccer",
          ],
          "type": "label",
        },
        Object {
          "instance": null,
          "key": "childcare_possible_answer-103476",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                checked={false}
                disabled={false}
                name={undefined}
                onBlur={undefined}
                onChange={[Function]}
                type="checkbox"
                value="childcare_possible_answer-103476"
              />,
              "Basketball",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "checked": false,
                "disabled": false,
                "name": undefined,
                "onBlur": undefined,
                "onChange": [Function],
                "type": "checkbox",
                "value": "childcare_possible_answer-103476",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            "Basketball",
          ],
          "type": "label",
        },
        Object {
          "instance": null,
          "key": "childcare_possible_answer-103477",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                checked={false}
                disabled={false}
                name={undefined}
                onBlur={undefined}
                onChange={[Function]}
                type="checkbox"
                value="childcare_possible_answer-103477"
              />,
              "Swimming",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "checked": false,
                "disabled": false,
                "name": undefined,
                "onBlur": undefined,
                "onChange": [Function],
                "type": "checkbox",
                "value": "childcare_possible_answer-103477",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            "Swimming",
          ],
          "type": "label",
        },
        Object {
          "instance": null,
          "key": "childcare_possible_answer-103478",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                checked={false}
                disabled={false}
                name={undefined}
                onBlur={undefined}
                onChange={[Function]}
                type="checkbox"
                value="childcare_possible_answer-103478"
              />,
              "Football",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "checked": false,
                "disabled": false,
                "name": undefined,
                "onBlur": undefined,
                "onChange": [Function],
                "type": "checkbox",
                "value": "childcare_possible_answer-103478",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            "Football",
          ],
          "type": "label",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
